src/tests/roc_ctl/test_task_queue.cpp

```diff
@@ -1290,5 +1290,53 @@ TEST(task_queue, reschedule_cancelled) {
     UNSIGNED_LONGS_EQUAL(1, tq.num_tasks());
 }
 
+TEST(task_queue, no_starvation) {
+    TestTaskQueue tq;
+    CHECK(tq.valid());
+
+    enum { NumTasks = 20 };
+
+    UNSIGNED_LONGS_EQUAL(0, tq.num_tasks());
+
+    TestHandler handler;
+    handler.expect_success(true);
+    handler.expect_n_calls(NumTasks);
+
+    TestTaskQueue::Task* tasks = new TestTaskQueue::Task[NumTasks];
```

Comment:
Since NumTasks is constant, we can use fixed-size array on stack.

Reply:
changed it to use fixed-size array on stack